components/RepositoryItem.tsx

```diff
@@ -25,10 +26,20 @@ export const RepositoryItem = ({ repository }: RepositoryItemProps) => {
   };
   const lastModified = useLastModified(repository.last_modified);
 
+  useEffect(() => {
+    if (isIssueOpen) {
+      setIsIssuesListVisible(true);
+    } else {
+      // Delay unmounting to allow close animation to complete
+      const timer = setTimeout(() => setIsIssuesListVisible(false), 300);
+      return () => clearTimeout(timer);
+    }
+  }, [isIssueOpen]);
+
   return (
     <div className="repo-item">
-      <div id={`repo-${repository.id}`} onClick={() => setIsIssueOpen(!isIssueOpen)}>
-        <div>
+      <div id={`repo-${repository.id}`}>
+        <div onClick={() => setIsIssueOpen(!isIssueOpen)}>
```

Comment:
moved `onClick` to this line so the `IssuesList` will be only closed if user clicked on the top-bar section of repo-item